### PR TITLE
Fix compose null value handling for container provisioner support

### PR DESCRIPTION
**Pull request by thrix**
When compose is set to null in YAML (or not provided), properly pass
null value to Testing Farm API instead of string "null". This enables
support for container provisioner with image specified in plan.

Changes:
- Update action logic to set environment.os = null when compose input is empty
- Modify schema to allow nullable compose values
- Update summary display to show "<container image from plan>" for null compose
- Add comprehensive test coverage for null compose behavior
- Update documentation with usage examples for compose: null

🤖 Generated with [Claude Code](https://claude.ai/code)

Co-Authored-By: Claude <noreply@anthropic.com>

### CODE REVIEW

**Comment by jamacku** [comment]
@thrix Thank you for the great PR! Please rebase to main and regenerate /dist files to resolve conflicts.

File: tests/action.test.ts [hunk 1]
```diff
@@ -1370,6 +1370,119 @@ describe('Integration tests - action.ts', () => {
...
+      });
+
+    // Run action
+    const octokit = new Octokit({ auth: 'mock-token' });
+    const pr = await PullRequest.initialize(new CustomContext(), octokit);
+
+    await action(pr);
+
+    // hardware request has not been called
+    expect(mocks.unsafeNewRequest).not.toHaveBeenCalled();
```
Comment (jamacku) [inline tests/action.test.ts:1462]: Please also check if we send null to TF

```suggestion
    expect(mocks.unsafeNewRequest).not.toHaveBeenCalled();
    
    expect(mocks.newRequest).toHaveBeenCalledWith(
      expect.objectContaining({
        environments: [
          expect.objectContaining({
            os: null, // os should be null
            arch: process.env['INPUT_ARCH'], // arch should be set
          }),
        ],
      }),
      false
    );
```

**Review by jamacku — CHANGES_REQUESTED** [review]
LGTM, just one minor nitpick.

**Comment by thrix** [comment]
@jamacku thanks, seems @phracek commited it, anything else?

**Review by jamacku — APPROVED** [review]
LGTM, Thank you.
